extensions/amp-user-notification/0.1/amp-user-notification.js

```diff
@@ -171,6 +188,22 @@ export class AmpUserNotification extends AMP.BaseElement {
     });
   }
 
+  /**
+   * Returns a promise that if user is in the given geoGroup
+   * @param {string} geoGroup
+   * @return {Promise<boolean>}
+   */
+  isNotificationRequiredGeo_(geoGroup) {
+    return Services.geoForDocOrNull(this.element).then(geo => {
+      user().assert(geo,
+          'requires <amp-geo> to use promptIfUnknownForGeoGroup');
+      return (
+        geo.ISOCountryGroups.indexOf(geoGroup) >= 0 ||
+        (geoGroup === '!countryGroup' && geo.ISOCountryGroups.length === 0)
```

Comment:
I am trying to think about the check to `geo.ISOCountryGroups.length === 0` here. 
A valid use case can be: One have 'eea' for show/hide notification, and also 'english' and 'spanish' for i18n. It's possible that one want to hide the user-notification even with `ISOCountryGroups.length > 0`.
What about supporting `data-show-if-geo=!${countryGroup}` instead?

Reply:
The reason I didn't got for !${countryGroup} is I wanted to be able to support the scenario where there are groups for nafta, eea and everybody else without having to list every country in the everybody else group.  A magic group name '!countryGroup' solves that.  The length === 0 means that no group matches and then the !countryGroup say in that case I want to trigger.  The example file does this.

There is a different case where there are multiple potentially overlapping notifications and the user can either create multiple, non-overlapping country groups or allow for stacked notifications according to taste.

Reply:
Removing `!countryGroup` and adding `data-show-if-not-geo`, also supporting multiple geos in one attribute. Per discussion with @zhouyx